File: 53. Maximum Subarray/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
  let max = nums[0], cur = 0;
  for(let i = 0; i < nums.length; i++){
    cur += nums[i];
    if(cur < 0 || cur < nums[i]) cur = nums[i];
    max = Math.max(cur, max);
  }
  return max;
};

File: 53. Maximum Subarray/solution.go
func maxSubArray(nums []int) int {
	if len(nums) == 0 { return 0 }
	var res, sum = math.MinInt32, 0
	for _, v := range nums {
			sum += v
			if sum > res { res = sum }
			if sum < 0 { sum = 0 }
	}
	
	return res
}